py/builtin.h

```diff
@@ -138,6 +138,9 @@ extern const mp_obj_module_t mp_module_sys;
 extern const mp_obj_module_t mp_module_errno;
 extern const mp_obj_module_t mp_module_uctypes;
 extern const mp_obj_module_t mp_module_machine;
+#if MICROPY_PY_MATH
```

Comment:
I don't think this guard is worth putting here.  The other modules just above don't have equivalent guards.

Reply:
> I usually put the same `#if` in the declaration as in the definition file so that undefined symbols are caught at compilation time rather than at link time

Yes, that is done in some places in the MicroPython code base.

But here IMO it's better to stick with local coding style to get local consistency.